### ISSUE TITLE
Use of labelled `break` in `if` in `inline for` loop in `switch` causes segfault

### ISSUE THREAD

**Issue by tgschultz** [ziglang/zig#7364]
```zig
test "" {
    const arr = [_]u8{0,1,2,3,4,5};
    var x = @as(u8, 1);
    const y = switch(x) {
        0 => 0,
        1 => blk: {
            inline for(arr) |z| {
                if(z == x) break :blk z;
            }
            unreachable;
        },
        else => 2,
    };
}
```

Removing any of the conditions listed in the title seems to fix it.

I believe the following may be related: #5426 #3029 #3136 #5579 which suggests this bug may be at least a year old.

**Comment by tiljeset** [comment]
I can confirm this issue, but I don't need the switch.

```zig
test "" {
    var x: u32 = 2;
    const arr = .{ 0, 1, 2, 3 };
    var valid = blk: {
        inline for (arr) |z| {
            if (z == x) break :blk true;
        }
        break :blk false;
    };
}
```

Tested on zig 0.8.0-dev.1140+9270aae07 for Linux x86_64.

**Comment by pancelor** [comment]
I just hit a segfault with something very similar:

```zig
const std = @import("std");

test "" {
    std.debug.print("running\n", .{});
    var items = [_]u8{ 0, 99, 0, 0 };

    var qi: usize = 0;
    search: while (qi < items.len - 1) : (qi += 1) {
        inline for (.{ 0, 1 }) |dir| {
            const n = items[qi + dir];
            if (n > 0) break :search;
        }
    } else {
        std.debug.print("else\n", .{});
    }
}
```

> $ zig version
> 0.10.0-dev.423+40c9ce2ca
> $ zig test src/labelcrash.zig
> Semantic Analysis [1003/1318] Segmentation fault

The segfault seems to disappear if I remove any of the following: the `if`, the `inline for` (manually inlining it), the `else` block (attached to the `while`)

**Comment by Vexu** [comment]
Fixed by 4618c41fa6ca70f06c7e65762d2f38d57b00818c
